.github/scripts/implement_agent.py

```
@@ -18,45 +19,85 @@
 def tool_bash(cmd: str) -> str:
+    """
+    Run a shell command and return its combined output, stderr, and exit code as a single string.
+    
+    Parameters:
+    	cmd (str): Shell command to execute.
+    
+    Returns:
+    	result (str): A string containing the command's stdout (truncated to the last 6000 characters), followed by "[stderr]" and the command's stderr (truncated to the last 2000 characters), and ending with "[exit N]" where N is the process exit code. If an exception occurs while running the command, returns a string in the format "[error: <exception>]".
+    """
     try:
         result = subprocess.run(cmd, shell=True, capture_output=True, text=True, timeout=120)
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)